source/Firmata/UwpFirmata.cpp

```diff
@@ -234,7 +238,156 @@ UwpFirmata::processInput(
     void
     )
 {
-    return ::RawFirmata.processInput();
+    uint16_t data = _firmata_stream->read();
+    if( data == static_cast<uint16_t>( -1 ) ) return;
+    
+    uint8_t byte = data & 0x00FF;
+    uint8_t upper_nibble = data & 0xF0;
+    uint8_t lower_nibble = data & 0x0F;
+
+    /*
+     * the relevant bits in the command depends on the value of the data byte. If it is less than 0xF0 (start sysex), only the upper nibble identifies the command
+     * while the lower nibble contains additional data
+     */
+    Command command = static_cast<Command>( ( data < static_cast<uint16_t>( Command::START_SYSEX ) ? upper_nibble : byte ) );
+
+    //determine the number of bytes remaining in the message
+    size_t bytes_remaining = 0;
+    bool isMessageSysex = false;
+    switch( command )
+    {
+    default: //command not understood
+    case Command::END_SYSEX: //should never happen
+        return;
+
+        //commands that require 2 additional bytes
+    case Command::DIGITAL_MESSAGE:
```

Comment:
A digital message is sent to the host to inform it that a value has changed. None of the input modes would work if we ignore these message types. My implementation ignores the messages a host wouldn't care about.